modules/programs/rofi-power.nix

```
@@ -0,0 +1,112 @@
+{ config, lib, pkgs, ... }:
+
+# Largely based on adnan360's power.sh gist:
+# https://gist.github.com/adnan360/f86012baeb4c9ca4f1af033550b03033
+#
+# Depends on rofi
+
+with lib;
+
+let
+  cfg = config.programs.rofi.power;
+  rofi-power = { states, logoutCmd }:
+    pkgs.writeScriptBin "rofi-power" ''
+      #!${pkgs.stdenv.shell}
+      chosen=$(cat <<EOF | rofi -dmenu -i
+      ${states}
+      EOF
+      )
+
+      if [[ $chosen == "Logout" ]]; then
+        ${logoutCmd}
+      elif [[ $chosen == "Shutdown" ]]; then
+        systemctl poweroff
+      elif [[ $chosen == "Reboot" ]]; then
+        systemctl reboot
+      elif [[ $chosen == "Suspend" ]]; then
+        systemctl suspend
+      elif [[ $chosen == "Hibernate" ]]; then
+        systemctl hibernate
+      elif [[ $chosen == "Hybrid-sleep" ]]; then
+        systemctl hibernate
+      elif [[ $chosen == "Suspend-then-hibernate" ]]; then
+        systemctl suspend-then-hibernate
+      fi
+    '';
+
+in {
+  meta.maintainers = [ maintainers.seylerius ];
+
+  options = {
+    programs.rofi.power = {
+      enable = mkEnableOption "Power menu based on rofi and systemd";
+
+      logoutCommand = mkOption {
+        default = "";
+        type = lib.types.str;
```

It is unnecessary to use `lib.` when the file starts with `with lib;`
```suggestion
        type = types.str;
```